### DISCUSSION TITLE
Is TLS 1.2 supported?

### DISCUSSION THREAD

**Discussion by Bernard314** [bertmelis/espMqttClient#50] [General]
Is TLS1.2 supported in this?

**Comment by bertmelis** [comment]
Yes, espMqttClient is built on WiFiClientSecure from the Arduino framework. TLS 1.2 is supported on both ESP32 and ESP8266, albeit not all cypher suites might be supported.

I'm running it with a local mosquitto broker with TLS 1.2

**Reply by bertmelis** [reply]
Just to add, the async version doesn't have TLS support at all. You might get it to work with certain forks but I haven't added support for async TLS to this library.

**Comment by bertmelis** [comment]
If you have trouble connecting, you should raise the debug level (are you using Arduino IDE or  Platformio or something else?).

Paste a log so we can check what is wrong.

**Reply by Bernard314** [reply]
Ok thanks. Still couldn't solve it so i posted my log in a separate issue #52
